pkg/version/legacy_examples/example_client.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2016 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package legacy_examples
+
+// An ExampleClient should use libcni to invoke a network plugin.
+// It should call ADD and DELETE, verifying all intermediate steps
+// and data structures.
+type ExampleClient struct {
+	Example
+}
+
+// V010_Client creates a simple ptp network configuration, then
+// executes libcni against the currently-built plugins.
+var V010_Client = ExampleClient{
+	Example{
+		Name:          "example_invoker_v010",
+		CNIRepoGitRef: "c0d34c69", //version with ns.Do
+		PluginSource: `package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"net"
+
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/libcni"
+)
+
+func main(){
+	code :=	exec()
+	os.Exit(code)
+}
+
+func exec() int {
+	confStr := "{ \"name\": \"default\", \"type\":\"ptp\", \"ipam\":{ \"type\": \"host-local\", \"subnet\": \"10.1.2.0/24\"}}"
+	netConf, err := libcni.ConfFromBytes([]byte(confStr))
+	if err != nil {
+		fmt.Printf("could not parse netconfig: %+v", err)
+		return 1
+	}
+
+	targetNs, err := ns.NewNS()
+	if err !=  nil {
+		fmt.Printf("Could not create ns: %+v", err)
+		return 1
+	}
+	defer targetNs.Close()
+
+	ifName := "eth0"
+
+	runtimeConf := &libcni.RuntimeConf{
+		ContainerID: "some-container-id",
+		NetNS:       targetNs.Path(),
+		IfName:      ifName,
+	}
+
+	cniConfig := &libcni.CNIConfig{Path: strings.Split(os.Getenv("PATH"), ":")}
+
+	result, err := cniConfig.AddNetwork(netConf, runtimeConf)
+	fmt.Printf("Result: %+v", result)
+	if err != nil {
+		fmt.Printf("noop add failed: %+v", err)
```

Comment:
this isn't the no-op plugin, right?  maybe it could read `cni add failed` instead?